e2e/_suites/fleet/services.go

```diff
@@ -270,32 +385,196 @@ func newDebianInstaller() (ElasticAgentInstaller, error) {
-		logDir:            "/var/lib/elastic-agent/data/elastic-agent-%s/logs/",
+		InstallFn:         installFn,
+		installerType:     "deb",
 		logFile:           "elastic-agent-json.log",
 		name:              binaryName,
 		path:              binaryPath,
-		PostInstallFn:     fn,
+		PostInstallFn:     postInstallFn,
+		PreInstallFn:      preInstallFn,
 		processName:       ElasticAgentProcessName,
 		profile:           profile,
 		service:           service,
 		tag:               tag,
+		UninstallFn:       unInstallFn,
+		workingDir:        binDir + "logs/",
 	}, nil
 }
 
+// newTarInstaller returns an instance of the Debian installer
+func newTarInstaller(image string, tag string) (ElasticAgentInstaller, error) {
+	image = image + "-systemd" // we want to consume systemd boxes
+	service := image
+	profile := FleetProfileName
+
+	// extract the agent in the box, as it's mounted as a volume
+	artifact := "elastic-agent"
+	version := agentVersion
+	os := "linux"
+	arch := "x86_64"
+	extension := "tar.gz"
+
+	tarFile, binaryPath, err := downloadAgentBinary(artifact, version, os, arch, extension)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"artifact":  artifact,
+			"version":   version,
+			"os":        os,
+			"arch":      arch,
+			"extension": extension,
+			"error":     err,
+		}).Error("Could not download the binary for the agent")
+		return ElasticAgentInstaller{}, err
+	}
+
+	commitFile := ".elastic-agent.active.commit"
+	homeDir := "/elastic-agent/"
+	binDir := "/usr/bin/"
+
+	preInstallFn := func() error {
+		commitFile := homeDir + commitFile
+		return installFromTar(profile, image, service, tarFile, commitFile, artifact, version, os, arch)
+	}
+	installFn := func(containerName string, token string) error {
+		hash, err := getElasticAgentHash(containerName, homeDir+commitFile)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"containerName": containerName,
+				"error":         err,
+			}).Error("Could not get agent hash in the container")
+
+			return err
+		}
+
+		// install the elastic-agent to /usr/bin/elastic-agent using command
+		binary := fmt.Sprintf("/elastic-agent/data/elastic-agent-%s/", hash) + artifact
+		args := []string{"--force", "--insecure", "--enrollment-token", token, "--kibana-url", "http://kibana:5601"}
+
+		err = runElasticAgentCommand(profile, image, service, binary, "install", args)
+		if err != nil {
+			return fmt.Errorf("Failed to install the agent with subcommand: %v", err)
+		}
+		return nil
+	}
+	enrollFn := func(token string) error {
+		args := []string{"http://kibana:5601", token, "-f", "--insecure"}
+
+		return runElasticAgentCommand(profile, image, service, ElasticAgentProcessName, "enroll", args)
+	}
+	postInstallFn := func() error {
+		log.Trace("No postinstall commands for TAR installer")
+		return nil
+	}
+	unInstallFn := func() error {
+		args := []string{"-f"}
+
+		return runElasticAgentCommand(profile, image, service, ElasticAgentProcessName, "uninstall", args)
```

Comment:
i believe ElasticAgentProcessName is `elastic-agent`, this should not be only process name but `/opt/Elastic/Agent/elastic-agent`

Reply:
I would expect running the installed linux service `elastic-agent` from PATH

Reply:
With https://github.com/elastic/e2e-testing/pull/378/commits/017e4d0b203147af7357f218ab7dbe9c38647de9 we are fixing the uninstalling scenario